src/script/index.js

```diff
@@ -24,10 +26,58 @@ for (var i = 0; i < paramPairsLength; i++) {
 delete i
 delete paramPair
 
+function add_instance(text, disabled, selected) {
+    const opt = document.createElement('option');
+    opt.innerText = text;
+    if (disabled) {
+        opt.setAttribute('disabled', true);
+    }
+    if (selected) {
+        opt.setAttribute('selected', true);
+    }
+    choose_instance.appendChild(opt);
+}
+
 if (prefillInstance != null) {
-    document.getElementById('instance').value = normalizeUrl(prefillInstance);
+    const url = normalizeUrl(prefillInstance);
+    instance.value = url;
+    add_instance(url.slice(8, url.length - 1), false, true);
 }
 
+choose_instance.addEventListener('focus', function (e) {
+    if (choose_instance.children.length < 3) {
+        if (choose_instance.children.length > 1) {
+            choose_instance.children[1].remove();
+        }
+        add_instance("... loading ...", true);
+        fetch('https://api.joinmastodon.org/servers')
+          .then(response => {
+              if (!response.ok) {
+                  throw new Error('Failure response from joinmastodon.');
+              }
+              return response.json();
+          })
+          .then(servers => {
+              choose_instance.children[1].remove();
+              const domains = servers.map(obj => obj.domain);
+              domains.sort();
+              for (const domain of domains) {
+                  add_instance(domain);
+              }
+          })
+          .catch(error => {
+              choose_instance.children[1].remove();
+              add_instance("--LOADING FAILED--", true);
+              console.error(
+                'Failed to fetch servers list from joinmastodon.', error);
+          });
```

Comment:
Since I am trying to pursue the IE 11 compatibility, it'd be nice to use `XMLHttpRequest` instead of `fetch` or optionally loading a polyfill for IE somehow

Reply:
I have no idea whether the other code I wrote works in IE 11, but I'll switch this to XMLHttpRequest.

Reply:
Resolved by 7850cece523515e4a0d70059804bf633402ab738